suite/tests/client-interface/drx-scattergather-aarch64.cpp

```diff
@@ -540,16 +636,54 @@ class input_data_t {
         // The base address is set to the 8th element in the region.
         return (1024 * offset) + (static_cast<size_t>(element_size) * 8);
     }
+};
 
-    template <typename T>
-    void
-    write_input_data(size_t offset, const std::array<T, 40> &input_data)
+class output_data_t : public test_memory_t {
+public:
+    output_data_t()
+        : test_memory_t()
     {
-        memcpy(region_start_addr(offset), input_data.data(),
-               input_data.size() * sizeof(T));
+        /*
+         * We set up nine pages of memory to use as input data for load instruction
+         * tests. The first page contains input data of different sizes and the rest
+         * of the pages are set up to fault when they are accessed. The tests use
```

Comment:
Do we want to make the bad pages readable but not writable, as these are stores?  Or it won't matter for our purposes?

Reply:
I don't think it really matters for the test. The faulting pages are there so we can write tests which test the faulting behaviour of the emulated instructions, but I haven't written those tests yet because we don't support state restore yet. When I do write those tests it doesn't really matter what the exact permissions are as long as it faults.